packages/eventstream-serde-universal/src/clientContextParams-precedence.integ.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { describe, expect, test as it } from "vitest";
+import { XYZService } from "xyz";
+
+describe("client context parameters precedence integration test", () => {
+  it("should handle conflicting vs non-conflicting parameter precedence correctly", async () => {
+    // For non-conflicting parameter root level takes precedence over nested defaults
+    const clientWithNonConflicting = new XYZService({
+      endpoint: "https://localhost",
+      apiKey: async () => ({ apiKey: "test-key" }),
+      customParam: "user-custom-value",
+      clientContextParams: {
+        apiKey: "test-key",
+        customParam: "default-custom-value",
+      },
+    });
+
+    // Verify the resolved config has the user's root-level value
+    const resolvedConfig = (clientWithNonConflicting as any).config;
+    expect(resolvedConfig.customParam).toBe("user-custom-value");
```

Comment:
This should be "nested-custom-value". When a value exists in both places the nested value should be used.